src/storage/txn/actions/prewrite.rs

```diff
@@ -412,6 +437,78 @@ impl<'a> PrewriteMutation<'a> {
         .into())
     }
 
+    fn check_assertion<S: Snapshot>(
+        &self,
+        reader: &mut SnapshotReader<S>,
+        write: &Option<(Write, TimeStamp)>,
+        write_loaded: bool,
+    ) -> Result<(Option<(Write, TimeStamp)>, bool)> {
+        if self.assertion == Assertion::None
+            || self.txn_props.assertion_level == AssertionLevel::Off
+        {
+            return Ok((None, false));
+        }
+
+        let mut reloaded_write = None;
+        let mut reloaded = false;
+
+        // To pass the compiler's lifetime check.
+        let mut write = write;
+
+        if write_loaded
+            && write.as_ref().map_or(
+                false,
+                |(w, _)| matches!(w.gc_fence, Some(gc_fence_ts) if !gc_fence_ts.is_zero()),
+            )
+        {
+            // The previously-loaded write record has an invalid gc_fence. Regard it as none.
+            write = &None;
+        }
+
+        // Load the most recent version if prev write is not loaded yet, or the prev write is not
+        // a data version (`Put` or `Delete`)
+        let need_reload = !write_loaded
```

Comment:
Do we need to reload if it is an invalid write due to existing gc_fence? It seems to me that it should be equivalent to a delete?🤔
Then, it looks in this case, the assertion would be incorrect after reload, because we treat a deleted (but gc-ed) entry as a existing one.

Reply:
If the current write has an invalid gc_fence, I expect `write_loaded` is true and `need_reload` is false. Does this make sense or is anything wrong in the code?

Reply:
Ah, the problem here is I thought `seek_write` is used later to reload the write. But actually, `get_write_with_commit_ts` is used, so the invalid gc_fence is ignored.